### PR TITLE
Documentation and release prep

### LINKED ISSUE #6: Readme / Metadata module misc
setup module structure and data

### PR DESCRIPTION
**Pull request by MartyEwings**
resolves #6

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+The role is intentonally left without permissions, and should be given only the permissions the installing organisation are authorised to grant to Puppet Enterprise Support personnel. For more information on RBAC permissions please see the [Puppet Enterprise Documentation](https://puppet.com/docs/pe/2019.8/rbac_permissions_intro.html)
+
+### Puppet Enterprise Database Access	
+
+The RSAN Platform has a Postgresql client installed, and is granted certificate based access to all Puppet Enterprise Databases on any pe_postgresl node within the current deployment. The access is limited to the [SELECT](https://www.postgresql.org/docs/11/sql-grant.html) privilege and is therefore READONLY in nature.
+
+To use this function execute the following command from the CLI of the RSAN host
+
+```
+psql "sslmode=verify-ca sslrootcert=/etc/puppetlabs/puppet/ssl/certs/ca.pem  sslcert=/etc/puppetlabs/puppet/ssl/certs/<fqdn_rsan_host>.pem sslkey=/etc/puppetlabs/puppet/ssl/private_keys/<fqdn_rsan_host>.pem hostaddr=<pe_postgres_host> port=5432 user=rsan dbname=<pe_db_name>"
```
Comment (jarretlavallee) [inline README.md]: For copy/paste ability

```suggestion
psql "host=$(puppet config print server) port=5432 user=rsan sslmode=verify-full sslcert=$(puppet config print hostcert) sslkey=$(puppet config print hostprivkey) sslrootcert=$(puppet config print localcacert) dbname=<pe_db_name>"
```

File: README.md [hunk 2]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+ - The RSAN importer class should only be applied one agent node
+ - All features are currently 
+
+## Known Issues
+
+- When accessing the database from the RSAN node for the first time, an error message will be presented: [#40](https://github.com/MartyEwings/RSAN/issues/40)
+
+```psql: private key file "/etc/puppetlabs/puppet/ssl/private_keys/rsan-target.platform9.puppet.net.pem" has group or world access; permissions should be u=rw (0600) or less```
+
+ To workaround change the private key file to 0600 or less, Puppet will manage this file back to incorrect permissions, for long term access making a copy of the key with the correct permissions is the best course of action.
```
Comment (jarretlavallee) [inline README.md:186]: Potential workaround: https://puppet.com/docs/puppet/7.1/config_file_main.html#special-types-of-values-for-settings

> Settings that take a single file or directory (like ssldir) can accept an optional hash of permissions. When starting up, Puppet enforces those permissions on the file or directory.

Perhaps we can set that for the `privatekeydir` or `hostprivkey` setting.

File: README.md [hunk 3]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+
+- When accessing the database from the RSAN node for the first time, an error message will be presented: [#40](https://github.com/MartyEwings/RSAN/issues/40)
+
+```psql: private key file "/etc/puppetlabs/puppet/ssl/private_keys/rsan-target.platform9.puppet.net.pem" has group or world access; permissions should be u=rw (0600) or less```
+
+ To workaround change the private key file to 0600 or less, Puppet will manage this file back to incorrect permissions, for long term access making a copy of the key with the correct permissions is the best course of action. 
+
+ - PuppetDB Metric Collection fails due to CVE-2020-7943  [27](https://github.com/MartyEwings/RSAN/issues/27)
 
+Please refer to the documenation of Puppet Metrics collector for recommended work arounds
```
Comment (jarretlavallee) [inline README.md:190]: This would be the puppet metrics dashboard module for this deployment.

File: README.md [hunk 4]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
 ## Usage
+The following outlines the main features of RSAN and how to consume them
+### Live Telemetry Display
+<<<<<<< HEAD
+
+The Rsan node will host an instance of the [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard)
+ 
+The Dashboard can be accessed on
+
+<RSAN-ip\>:3000\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
http://<RSAN-ip\>:3000\
```

File: README.md [hunk 5]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
-puppet-grafana (>= 3.0.0 < 7.0.0)
-puppet-telegraf (>= 2.0.0 < 4.0.0)
-puppetlabs-apt (>= 4.3.0 < 8.0.0)
-puppetlabs-inifile (>= 2.0.0 < 5.0.0)
-puppetlabs-puppetserver_gem (>= 1.1.1 < 3.0.0)
-puppet/openvpn (>= 8.3.0)
 
+Module Dependencies
 
+ - derdanne/nfs (>= 2.1.5)
```
Comment (jarretlavallee) [inline README.md]: The formatting is inconsistent with these new ones.

```suggestion
- derdanne/nfs (>= 2.1.5)
```

**Review by jarretlavallee — CHANGES_REQUESTED** [review]
Looks pretty good. I put in a few nit picks. You may also want to generate the REFERENCE.md. The code headers may need to be updated to account for it. 

```
pdk bundle exec rake strings:generate:reference
```

The REFERENCE.md will populate the reference page in the forge and generate docs around the code from the header comments with `puppet-strings.

File: README.md (hunk 5 above)
Comment (jarretlavallee) [reply README.md]: To solve the formatting issue, you may need to add the `  -` to these.

File: README.md (hunk 3 above)
Comment (MartyEwings) [reply README.md:190]: doh

File: README.md (hunk 5 above)
Comment (jarretlavallee) [reply README.md]: Nevermind. It would just be one or the other. The unordered list works well.

File: README.md (hunk 3 above)
Comment (jarretlavallee) [reply README.md:190]:
```suggestion
Please refer to the documentation of [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard) for recommended workarounds
```

**Comment by jarretlavallee** [comment]
We should document resource requirements so the customer knows what size of a machine needs to be deployed for RSAN.

**Comment by jarretlavallee** [comment]
Also, the agent OS needs to be the same OS as the master or it cannot deploy the client tools automatically.

File: README.md [hunk 6]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+
+  - Run Puppet on all nodes in "PE Infrastructure Agent" node group
+
+   - Decommission the RSAN platform 
 
-TBC - detailed description of feature switches and configurable parameters
 
 ## Limitations
+ - The RSAN importer class should only be applied one agent node
+ - All features are currently
```
Comment (jarretlavallee) [inline README.md]: Currently what?

Comment (MartyEwings) [reply README.md]: doh!

File: README.md [hunk 7]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+
+/var/pesupport/<FQDN of Infrastructure node\>/var/log\
+/var/pesupport/<FQDN of Infrastructure node\>/opt/puppetlabs\
+/var/pesupport/<FQDN of Infrastructure node\>/etc/puppetlabs
+
+#### Optional Configuration
+
+The RSAN Class assumes the RSAN server will mount the shared partitions using the IP address Source designated by the "ipaddress" fact. In any deployment should this assertion not be true, it is nessary to set the following parameter to the source IP address of the RSAN Host:
+
+**rsan::exporter::rsan_importer_ips**
```
Comment (jarretlavallee) [inline README.md]: This should show the syntax of setting this parameter or link to the REFERENCE.md for information.

In hiera"

```
rsan::exporter::rsan_importer_ips:
  - 1.2.3.4
 ```

In the class declaration or in the console the value would be an array

```
["1.2.3.4"]
```

File: README.md [hunk 8]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+User: admin\
+Password: admin
+
+For advanced configuration and documentation please see [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard)
+
+### Infrastructure node file and log access	
+
+The RSAN node will, by default, mount /var/log/ /opt/puppetlabs and /etc/puppetlabs from each of the Puppet Enterprise Infrastructure nodes on the RSAN platform in the following location, as read only file systems.
+
+/var/pesupport/<FQDN of Infrastructure node\>/var/log\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
`/var/pesupport/<FQDN of Infrastructure node\>/var/log\`
```

File: README.md [hunk 9]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+/var/pesupport/<FQDN of Infrastructure node\>/opt/puppetlabs\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
`/var/pesupport/<FQDN of Infrastructure node\>/opt/puppetlabs\`
```

File: README.md [hunk 10]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+/var/pesupport/<FQDN of Infrastructure node\>/etc/puppetlabs
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
`/var/pesupport/<FQDN of Infrastructure node\>/etc/puppetlabs`
```

File: README.md [hunk 11]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
 ## Usage
+The following outlines the main features of RSAN and how to consume them
+### Live Telemetry Display
+<<<<<<< HEAD
+
+The Rsan node will host an instance of the [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard)
+ 
+The Dashboard can be accessed on
+
+http://<RSAN-ip\>:3000\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
**URL:** http://<RSAN-ip\>:3000\
```

File: README.md [hunk 12]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+User: admin\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
**User:** admin\
```

File: README.md [hunk 13]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+Password: admin
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
**Password:** admin
```

File: README.md [hunk 14]
```diff
@@ -30,43 +30,145 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+
+For advanced configuration and documentation please see [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard)
+
+### Infrastructure node file and log access	
+
+The RSAN node will, by default, mount /var/log/ /opt/puppetlabs and /etc/puppetlabs from each of the Puppet Enterprise Infrastructure nodes on the RSAN platform in the following location, as read only file systems.
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
The RSAN node will, by default, mount `/var/log/`, `/opt/puppetlabs` and `/etc/puppetlabs` from each of the Puppet Enterprise Infrastructure nodes on the RSAN platform in the following location, as read-only file systems.
```

**Review by jarretlavallee — COMMENTED** [review]
Minor formatting updates

**Comment by MartyEwings** [comment]
> We should document resource requirements so the customer knows what size of a machine needs to be deployed for RSAN.

iv been testing on m1 medium, and that's probably as small as you want any Puppet agent, ill set that as minimum

File: README.md (hunk 2 above)
Comment (MartyEwings) [reply README.md:186]: im going to see if i can resolve this in an issue

#40

File: README.md [hunk 15]
```diff
@@ -30,43 +30,166 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+**User:** admin\
+**Password:** admin
+
+For advanced configuration and documentation please see [Puppet Metrics Dashboard](https://forge.puppet.com/modules/puppetlabs/puppet_metrics_dashboard)
+
+### Infrastructure node file and log access	
+
+The RSAN node will, by default, mount `/var/log/`, `/opt/puppetlabs` and `/etc/puppetlabs` from each of the Puppet Enterprise Infrastructure nodes on the RSAN platform in the following location, as read-only file systems.
+
+`/var/pesupport/<FQDN of Infrastructure node\>/var/log\`
```
Comment (jarretlavallee) [inline README.md]: My bad on these. They need to be an unordered list and remove the `\>` now that they are in the inline code.

```
* `/var/pesupport/<FQDN of Infrastructure node>/var/log`
```

File: README.md [hunk 16]
```diff
@@ -30,43 +30,153 @@ Software required for the proper functioning of the RSAN will be deployed on the
...
+The RSAN node will deploy Puppet Client tools for use by Puppet Enterprise on the RSAN platform, For More information please see the Puppet Enterprise Documentation:
+
+[PE Client tools](https://puppet.com/docs/pe/2019.8/installing_pe_client_tools.html)
+
+A supplementary task is available to generate an RBAC user and role, so that the credentials may be used provided to Puppet Enterprise Support personnel.
+
+#### Creating Support User
+
+Run the following task against the Primary Puppet Enterprise Server\
+For imformation on executing PE tasks see the [Puppet Enterprise Documentation](https://puppet.com/docs/pe/2019.8/tasks_in_pe.html)\
```
Comment (jarretlavallee) [inline README.md]:
```suggestion
For information on executing PE tasks see the [Puppet Enterprise Documentation](https://puppet.com/docs/pe/2019.8/tasks_in_pe.html)\
```

**Review by jarretlavallee — APPROVED** [review]
LGTM ship it

**Pull request merged**